test/schemes/boundary/dummy_particles/dummy_particles.jl

```diff
@@ -188,6 +188,218 @@
+                                                      tank1.boundary.coordinates, particle)
+            end
+
+            @test all(boundary_system.boundary_model.pressure .== 0.0)
+            @test all(fluid_system1.pressure .== 0.0)
+        end
+
+        # Test whether the pressure is constant if the density of the state equation
+        # and in the tank are not the same.
+        # Then we test that the extrapolation yields some constant value.
+        @testset "Constant Non-Zero Pressure" begin
+            density = 260
+            tank2 = RectangularTank(particle_spacing, (width, height), (width, height),
+                                    density, n_layers=n_layers,
+                                    faces=(true, true, true, false))
+
+            fluid_system2 = WeaklyCompressibleSPHSystem(tank2.fluid, SummationDensity(),
+                                                        state_equation,
+                                                        smoothing_kernel, smoothing_length)
+
+            fluid_system2.cache.density .= tank2.fluid.density
+            v_fluid = zeros(2, TrixiParticles.nparticles(fluid_system2))
+            TrixiParticles.compute_pressure!(fluid_system2, v_fluid)
+
+            neighborhood_search = TrixiParticles.TrivialNeighborhoodSearch{2}(search_radius=TrixiParticles.compact_support(smoothing_kernel,
+                                                                                                                           smoothing_length),
+                                                                              eachpoint=TrixiParticles.eachparticle(fluid_system2))
+
+            TrixiParticles.set_zero!(boundary_model.pressure)
+            TrixiParticles.reset_cache!(boundary_system.boundary_model.cache,
+                                        viscosity)
+
+            TrixiParticles.boundary_pressure_extrapolation!(boundary_model, boundary_system,
+                                                            fluid_system2,
+                                                            tank2.boundary.coordinates,
+                                                            tank2.fluid.coordinates,
+                                                            v_fluid,
+                                                            v_fluid,
+                                                            neighborhood_search)
+
+            for particle in TrixiParticles.eachparticle(boundary_system)
+                TrixiParticles.compute_adami_density!(boundary_model, boundary_system,
+                                                      tank2.boundary.coordinates, particle)
+            end
+
+            @test all(isapprox.(boundary_system.boundary_model.pressure,
+                                boundary_system.boundary_model.pressure[1], atol=1.0e-12))
+            @test all(isapprox.(fluid_system2.pressure, fluid_system2.pressure[1]))
```

Comment:
```suggestion
            # Test that pressure of the fluid is indeed constant
            @test all(isapprox.(fluid_system2.pressure, fluid_system2.pressure[1]))
            # Test that boundary pressure equals fluid pressure
            @test all(isapprox.(boundary_system.boundary_model.pressure,
                                fluid_system2.pressure[1], atol=1.0e-12))
```
Right now, you are not testing that fluid and boundary pressure are the same.